Example-Crypto-REXX/RXCRYPT1.rexx

```diff
@@ -0,0 +1,2453 @@
+        return 0
+  end
+
+
+/* if  still here then something went wrong */
+return fatalError('ERROR ** failed to obtain LOCAL CPC info **')
+
+/*******************************************************/
+/* Function:  GetCPCInfo                               */
+/*                                                     */
+/* Retrieve the uri, target name and SE version        */
+/* associated with CPCname                             */
+/* Prime CPCuri and CPCtargetName variables            */
+/* Verify SE version is z16 or higher                  */
+/*                                                     */
+/* Return 0 if successful, otherwise return -1         */
+/*******************************************************/
+GetCPCInfo:
+
+methodSuccess = TRUE /* assume success */
+emptyCPCResponse = '{"cpcs":[]}'
+
+/* First list the cpcs, filtering the response for
+   just information regarding CPC
+
+   GET /api/cpcs?name=CPCname
+
+   v1: HWICONN(HWI_CPC,IBM390PS.CPC)
+
+   NOTE:
+   If you wanted to obtain information for the 'LOCAL'
+   CPC, then you'd retrieve the full list and search for
+   the CPC with the
+         "location":"LOCAL"
+   attribute, this would be equivalent to HWICONN(HWI_CPC,*)
+*/
+reqUri = '/api/cpcs?name='||CPCname
+CPCInfoResponse = GetRequest(reqUri)
+
+emptyCPCArray = INDEX(CPCInfoResponse, emptyCPCResponse)
+if emptyCPCArray > 0 | CPCInfoResponse = '' Then
+  do
+    return fatalError('** failed to retrieve CPC **')
+  end
+
+/* Parse the response to obtain the uri, target name
+   and SE version associated with the CPC 
+*/
+call JSON_parseJson CPCInfoResponse
+
+CPCversion = JSON_findValue(0,"se-version", HWTJ_STRING_TYPE) 
+if foundJSONValue(CPCversion) = FALSE then                    
+  methodSuccess = FALSE                                       
+
+CPCuri = JSON_findValue(0,"object-uri", HWTJ_STRING_TYPE)
+if foundJSONValue(CPCuri) = FALSE then
+  methodSuccess = FALSE
+
+CPCtargetName = JSON_findValue(0,"target-name", HWTJ_STRING_TYPE)
+if foundJSONValue(CPCtargetName) = FALSE then
+  methodSuccess = FALSE
+
+if methodSuccess then
+  do
+    Say
+    Say 'Successfully obtained CPC Info:'
+    Say '  uri:'||CPCuri
+    Say '  target-name:'||CPCtargetName
+    Say '  SE version: '||CPCversion
+    Say
+    if substr(CPCversion,3,2) < 16 then                                 
+      do                                                                 
+       errMsg = '** Crypto attributes supported on z16 or higher only **'
+       return fatalError(errMsg)                                
+      end                                                                
+      else                                                                
+       return 0                                                         
+  end
+
+Say
+Say 'Obtained some or none of the CPC Info:'
+Say '  uri:('||CPCuri||')'
+Say '  target-name:('||CPCtargetName||')'
+Say 'full response body:('||CPCInfoResponse||')'
+Say
```

Comment:
i suspect this would result in the same information twice for the case of methodSuccess = TRUE, i'd wrap this under an else corresponding to the if on 462

Reply:
i take it back...i missed the returns above..... meh..your call then

Reply:
you are correct, this belongs in an else